pyanaconda/storage/checker.py

```diff
@@ -411,6 +413,60 @@ def verify_mounted_partitions(storage, constraints, report_error, report_warning
                                "installation. Please unmount it and retry.") % part.path)
 
 
+def verify_lvm_destruction(storage, constraints, report_error, report_warning):
+    """Verify that destruction of LVM devices is correct.
+
+    This means: A VG and all its PVs must be all destroyed together, nothing is allowed to remain.
+
+    :param storage: a storage to check
+    :param constraints: a dictionary of constraints
+    :param report_error: a function for error reporting
+    :param report_warning: a function for warning reporting
+    """
+    # Implementation of the verification uses an inverted approach: Check that all PVs to be
+    # destroyed are contained in a VG that will be destroyed too. If there is a PV to be destroyed
+    # that has a VG that isn't going to be destroyed, that is an error. This is because we ask
+    # blivet to ignore some disks, and these might contain a PV. Crucially, blivet implements this
+    # such that the ignoring hides also all dependent devices - for any disk, it could be a PV
+    # partition, which would then hide also all VGs and LVs that depend on the PV. It does not
+    # matter that the VG needs other PVs too - the hiding wins.
+    destroyed_vg_names = set()
+    all_touched_vgs_and_disks = defaultdict(list)
+
+    for action in storage.devicetree.actions:
+        if action.is_destroy and action.is_device and action.device.type == "lvmvg":
+            destroyed_vg_names.add(action.device.name)
+
+        elif action.is_destroy and action.is_format and action.orig_format.type == "lvmpv":
+            # Check if the PV actually had any VG assigned
+            if action.orig_format.vg_name:
+                pv_name = action.device.name
+                disk_name = action.device.disk.name
+                vg_name = action.orig_format.vg_name
+                all_touched_vgs_and_disks[vg_name].append(disk_name)
+
+    log.debug("verify_lvm_destruction : destroyed_vg_names, all_touched_vgs_and_disks :")
+    log.debug(destroyed_vg_names)
+    log.debug(all_touched_vgs_and_disks)
+    for vg_name in set(all_touched_vgs_and_disks):
+        if vg_name not in destroyed_vg_names:
+            selected_disks = all_touched_vgs_and_disks[vg_name]
+            helper_disk = storage.devicetree.get_device_by_name(selected_disks[0])
+            related_disks = storage.devicetree.get_related_disks(helper_disk) # THIS IS DOING WRONG THINGS
```

Comment:
I don't think this is going to work, because the destroyed LVM devices that define the relations between disks are not in the device tree at this moment.

Reply:
Yep, Vojta confirms that the function is not going to work once actions happen. The untouched disks are linked via the to-be-destroyed LVM bits, while the touched disks via the to-be-created bits. That's impossible to reconcile, so in the end it spews nonsense.